language/move-lang/tests/move_check/naming/generics_shadowing_invalid.exp

```diff
@@ -1,3 +1,14 @@
+error[E04007]: incompatible types
+  ┌─ tests/move_check/naming/generics_shadowing_invalid.move:7:14
+  │
+6 │     fun foo<S>(s1: S, s2: S): S {
+  │                    - The type: 'S'
+7 │         (s1: Self::S);
+  │              ^^^^^^^
+  │              │
+  │              Invalid type annotation
+  │              Is not compatible with: '0x2::M::S'
+
```

Comment:
Not sure if you want to change this in this PR or not, but the ordering of these error messages are a bit funky to me. IMO If I read them from top to bottom, it reads out of order.

I want to read this as "Invalid type annotation" -> "The type: 'S'" -> "Is not compatible with: '0x2::M::S'"

Reply:
You should no longer think about these being read from top to bottom, that's why this refactoring has been put off for so long ;) 

With the new codespan reporting, it is designed to be a bit more IDE friendly, as such they are sorted based on the textual location, not based on the order provided. With colors, it comes out a bit cleaner (hopefully).

Here are the errors for this test
![image](https://user-images.githubusercontent.com/1753366/125718595-8c6da387-cdb8-46f1-97ff-be1f728be1d8.png)

Reply:
Makes a lot more sense with colors!